backend/src/Logging.hs

```diff
@@ -52,6 +53,17 @@ fileLogger path = do
   (logger, _) <- newTimedFastLogger timeCache (LogFileNoRotate path defaultBufSize)
   pure $ \_ _ level msg -> logger (\time -> formatLogMsg time level msg)
 
+fromAwsLogLevel :: AWS.LogLevel -> LogLevel
+fromAwsLogLevel AWS.Trace = LevelDebug
+fromAwsLogLevel AWS.Debug = LevelDebug
+fromAwsLogLevel AWS.Info = LevelInfo
+fromAwsLogLevel AWS.Error = LevelError
+
+toAwsLogger :: LogLevelFilter -> Logger -> AWS.Logger
+toAwsLogger logFilter logger awsLevel builder = when (logFilter "" logLevel) (log logger logLevel (toLogStr builder))
```

Comment:
No productive feedback, just learning things! Was thrown because the function signature looks like it accepts 2 args, but there are 4 params?? And then looked at the types and realized AWS.Logger is a type of function that accepts 2 params, so when `toAwsLogger logFilter logger` is called in `Server.hs`, the result is a partially applied function matching the AWS.Logger function type. My Haskell lesson for the day ✔️

Reply:
Yep! Basically just one of many ways to write a function that returns a function. I could have also written a two argument version that more explicitly defines and returns a different two argument function - perhaps that would have been clearer?

Nice work reading this and checking the type for more understanding when it looked weird. Your Haskell-fu has gotten so much stronger 💪

Reply:
Ohhh it's like a decorator, that didn't even click when I wrote it out. And it was written totally clearly IMO, sometimes no amount of clarity can fix a newbie! Yay, thank you! 🙇‍♀️ 🎉